[MAINTENANCE] build-gallery enhancements

kenwade4 opened the pull request:
Changes proposed in this pull request:
- Update build_test_backends_list func with redshift/athena/snowflake 
- Add trigger_algolia.sh script
- Update conftest options
- Update build_sa_validator_with_data to return engines for redshift/athena/snowflake
- Increase timeout of the build_gallery CI job
- Update get_dataset and build_sa_validator_with_data funcs with more types
    - Add REDSHIFT_TYPES, SNOWFLAKE_TYPES, and ATHENA_TYPES to self_check.util
- Update build_gallery.py to accept --outfile-name and --backends options
- Add force_no_progress_bar kwarg to validator/util methods; force off for diagnostics
- Comment out forcing sqlite if no other sql dialects selected for tests
- Allow a top-level "suppress_test_for" (to skip all tests for a backend)
- Update run_diagnostics to accept a debug_logger and time durations of slow parts
- For test_package.py only run_diagnostics on pandas/sqlite (fast backends)
- Add coverage_score to run_diagnostics
- Install great_expectations_contrib in the compatibility matrix job
- In dependency_graph, ignore contrib/cli tests for dgtest
- Add a section to show testing warnings in the expectation_gallery pipeline
- Add get_expectation_file_info_dict func to build_gallery.py script 
    - Get create/update times and Expectation type
    - Only process the Expectations that were specified

### Definition of Done


- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

kenwade4 committed:
Add skeleton of trigger_algolia.sh script and update azure-pipelines-expectation-gallery.yml

kenwade4 committed:
Update build_test_backends_list func with redshift/athena/snowflake

Also add creds to azure-pipelines-expectation-gallery.yml and re-attempt
bigquery in some situations

kenwade4 committed:
Temporarily change name of generated JSON files

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Update env var names in AlgoliaScripts README

kenwade4 committed:
Update conftest options

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5616&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5616&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/5616/f4ec4d07/8de26351b3d2c4c57d9a3378ffc9a85bef6da5a8.svg)](https://app.codesee.io/r/reviews?pr=5616&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

kenwade4 committed:
Reformat with black

kenwade4 committed:
Temporarily ignore bigquery again

kenwade4 committed:
Remove unecessary positional arg

kenwade4 committed:
Enable DownloadSecureFile tasks in expectation_gallery pipeline

kenwade4 committed:
Include bigquery in dialects_to_include

kenwade4 committed:
Update build_sa_validator_with_data to return enginers for redshift/athena/snowflake

kenwade4 committed:
Increase the timeout of the build_gallery CI job

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Temporarily comment out athena in SQL_DIALECT_NAMES tuple

kenwade4 committed:
Temporarily don't run on contrib expectations

kenwade4 committed:
Update get_dataset and build_sa_validator_with_data funcs w/ more types

Add REDSHIFT_TYPES, SNOWFAKE_TYPES, and ATHENA_TYPES to self_check.util

kenwade4 committed:
Add force_no_progress_bar kwarg to validator/util methods; force off for diagnostics

kenwade4 committed:
Comment out forcing sqlite if no other sql dialects selected for tests

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Allow existing redshit quantiles fix code to be run

kenwade4 committed:
Bump timeout for build_gallery to 3 hours

kenwade4 committed:
Update run_diagnostics to accept a debug_logger and time durations of slow parts

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Don't put all the duration strings on the same debug message

kenwade4 committed:
Update trigger_algolia.sh to call the node scripts by name

kenwade4 committed:
Bump build_gallery timeout to 5hrs to see how long it takes to include contrib too

kenwade4 committed:
Update S3Upload task after build_gallery to upload .txt files too

kenwade4 committed:
Uncomment athena from SQL_DIALECT_NAMES tuple

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Comment out import of dremio/teradata in self_check/util.py

kenwade4 committed:
Bump TYPE_HINT_ERROR_THRESHOLD

kenwade4 committed:
Don't run quantiles test for bigquery... df.to_sql took over an hour

kenwade4 committed:
Allow only_consider_these_backends list to be passed to run_diagnostics (and its relevant helpers)

kenwade4 committed:
Enable top-level suppress_test_for in test_definitions JSON files

kenwade4 committed:
Allow generate_expectation_tests to better alert/log when get_test_validator_with_data fails

kenwade4 committed:
Allow receiving extra_debug info and improve debug/error logging

kenwade4 committed:
Suppress a particular quantile test for redshift (slow)

kenwade4 committed:
Comment athena out of SQL_DIALECT_NAMES

kenwade4 committed:
Re-attempt to upload .json and .txt files to S3

kenwade4 committed:
Add several output analysis stages to expectation_gallery pipeline

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Only try to upload JSON files to S3

kenwade4 committed:
Only cat gallery-tracebacks.txt if the file exists

kenwade4 committed:
Try another way to specify bash expression in pipeline

kenwade4 committed:
Try another way to cat a file after ensuring it exists

kenwade4 committed:
Update call to build_gallery.py to not limit Expectations or backends to try

kenwade4 committed:
Add a build_gallery step to show entire checklist and another for partial checklist

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Install aws-glue-libs from github repo in docusaurus_tests stage

Prevent ModuleNotFoundError for awsglue

https://github.com/awslabs/aws-glue-libs/issues/3

kenwade4 committed:
For test_package.py only run_diagnostics on pandas/sqlite (fast backends)

kenwade4 committed:
Add a coverage_score to run_diagnostics

kenwade4 committed:
Add --no-strict-optional to type_check task

See:
    - https://mypy.readthedocs.io/en/stable/kinds_of_types.html#no-strict-optional
    - https://mypy.readthedocs.io/en/stable/common_issues.html

Also add `# type: ignore` to some problematic lines that give errors like these
    - error: Cannot assign to a type  [misc]
    - error: Argument 1 to "append" of "list" has incompatible type "float"; expected "int" [arg-type]
    - error: Redundant cast to "BatchDefinition"  [redundant-cast]

kenwade4 committed:
Install great_expectations_contrib in the compatibility_matrix job

Prevent ModuleNotFoundError for great_expectations_contrib in
tests/test_package.py

kenwade4 committed:
In dependency_graph, ignore contrib/cli/tests for dgtest

kenwade4 committed:
Use SparkContext.getOrCreate() in aws_glue_deployment_patterns.py

Causing ValueError: Cannot run multiple SparkContexts at once

See https://stackoverflow.com/a/47731444

kenwade4 committed:
Add a section to show testing warnings in the expectation_gallery pipeline

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Move some #noqa from decorator line to method def line

kenwade4 committed:
Ignore C901 flake8 errors (too complex) in setup.cfg

kenwade4 committed:
Try upping the max-complexity in setup.cfg

kenwade4 committed:
Add a 2hr timeout for test_docs job in docs_integration pipeline

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Add get_expectation_file_info_dict func to build_gallery.py script

kenwade4 committed:
Update 'show testing warning' step to not exit with an error code if there are no warnings

kenwade4 committed:
Show Expectation update/create time in build gallery log, and log output fixes

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Add step to install the contrib CLI tool

kenwade4 committed:
Don't try to cd to contrib/cli since the workingDirectory is specified already

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Try using the same workingDirectory so its available in the next step

kenwade4 committed:
Update get_expectation_file_info_dict to get the Expectation type(s)

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Update get_expectation_file_info_dict to only process what's needed

kenwade4 committed:
Generate and use the gallery-exp-types.txt file

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
Reformat with black

AFineDayFor commented on assets/scripts/trigger_algolia.sh:
```diff
@@ -0,0 +1,18 @@
+which yarn
+
+if [[ ! -d "AlgoliaScripts" ]]; then
+    echo "The AlgoliaScripts directory isn't in this branch yet"
```
Not even a NIT: lol, printf drives more people mad.

AFineDayFor commented on assets/scripts/trigger_algolia.sh:
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+which npm
```
NIT: if it isn't, it could die if these weren't present and save you some CPU time if something ever goes awry this early on.

kenwade4 replied:
I could actually get rid of the which statements. This is only called in an azure pipeline stage that is right after the "setup node" part

AFineDayFor commented on azure-pipelines-docs-integration.yml:
```diff
@@ -82,6 +82,7 @@ stages:
       vmImage: 'ubuntu-latest'
     jobs:
     - job: test_docs
+      timeoutInMinutes: 120
```
The default being 60, I'm fine with this change :bow:

kenwade4 replied:
Something crept up in one of the runs and the pipeline was killed at 60 minutes while it was still doing stuff

AFineDayFor commented on azure-pipelines-expectation-gallery.yml:
```diff
@@ -130,21 +134,100 @@ stages:
+              retryCount: '2'
 
-          - bash: python ./build_gallery.py
+          - bash: python ./build_gallery.py --outfile-name "expectation_library_v3.json" | tee output--build_gallery.txt ; grep -o "Took .* seconds to .*" output--build_gallery.txt | sort -k2,2nr > testing-times.txt ; grep -o "ERROR - (.*" output--build_gallery.txt | sort > testing-error-messages.txt; touch gallery-tracebacks.txt; grep -o "Expectation type.*" output--build_gallery.txt | sort > ga ...
```
much respec :) I really do like the validation here.

kenwade4 replied:
Yea it powers all those little summary/info sections below

AFineDayFor commented on great_expectations/expectations/expectation.py:
```diff
@@ -4,10 +4,11 @@
 import warnings
 from abc import ABC, ABCMeta, abstractmethod
-from collections import Counter
+from collections import Counter, defaultdict
```
🙇

AFineDayFor commented on great_expectations/expectations/expectation.py:
```diff
@@ -970,6 +975,10 @@ def run_diagnostics(
         that do not exist). These errors are added under "errors" key in the report.
         """
 
+        _debug = lambda x: x
```
big fan of better logging and reporting; thank you for this part in particular!

AFineDayFor reviewed (approved):
Appreciate the maint

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into misc/build-gallery-enhancements

kenwade4 committed:
When trying to import a contrib Expectation, don't die if something like a TypeError happens

kenwade4 committed:
Rename the json files to what they should be
